libs/barista-components/experimental/datepicker/src/timeinput.spec.ts

```diff
@@ -0,0 +1,291 @@
+ * limitations under the License.
+ */
+
+import {
+  NUMPAD_MINUS,
+  NUMPAD_ONE,
+  NUMPAD_PERIOD,
+  NUMPAD_PLUS,
+} from '@angular/cdk/keycodes';
+import { Component, ElementRef, ViewChild } from '@angular/core';
+import {
+  ComponentFixture,
+  fakeAsync,
+  flush,
+  TestBed,
+  tick,
+  waitForAsync,
+} from '@angular/core/testing';
+import {
+  DtDatepickerModule,
+  DtTimeInput,
+} from '@dynatrace/barista-components/experimental/datepicker';
+import {
+  createComponent,
+  dispatchFakeEvent,
+  dispatchKeyboardEvent,
+  typeInElement,
+} from '@dynatrace/testing/browser';
+
+describe('DtTimeInput', () => {
+  beforeEach(
+    waitForAsync(() => {
+      TestBed.configureTestingModule({
+        imports: [DtDatepickerModule],
+        declarations: [SimpleTimeInputTestApp],
+      });
+
+      TestBed.compileComponents();
+    }),
+  );
+
+  describe('basic behavior', () => {
+    let fixture: ComponentFixture<SimpleTimeInputTestApp>;
+    let component: SimpleTimeInputTestApp;
+    let hourEl: HTMLInputElement;
+    let minuteEl: HTMLInputElement;
+
+    beforeEach(() => {
+      fixture = createComponent(SimpleTimeInputTestApp);
+      component = fixture.componentInstance;
+      hourEl = component.timeInput._hourInput.nativeElement;
+      minuteEl = component.timeInput._minuteInput.nativeElement;
+      fixture.detectChanges();
+    });
+
+    describe('disabled input property', () => {
+      it('should not be usable when disabled', fakeAsync(() => {
+        component.disabled = true;
+        fixture.detectChanges();
+        tick();
+        expect(hourEl.disabled).toBeTruthy();
+        expect(hourEl.getAttribute('aria-disabled')).toBeTruthy();
+        expect(minuteEl.disabled).toBeTruthy();
+        expect(minuteEl.getAttribute('aria-disabled')).toBeTruthy();
+      }));
+    });
+
+    describe('timeChange event', () => {
+      it('should emit a timechange event when the hour and minute inputs are changed', fakeAsync(() => {
+        const changeSpy = jest.fn();
+        component.timeInput.timeChange.subscribe(changeSpy);
+        fixture.detectChanges();
+        expect(changeSpy).not.toHaveBeenCalled();
+        component.timeInput.hour = 23;
+        component.timeInput.minute = 55;
+        fixture.detectChanges();
+        dispatchFakeEvent(hourEl, 'blur');
+        expect(changeSpy).toHaveBeenCalledTimes(1);
+      }));
+
+      it('should emit a timechange event when the hour and minute inputs are reset to empty', fakeAsync(() => {
+        const changeSpy = jest.fn();
+        component.timeInput.timeChange.subscribe(changeSpy);
+        fixture.detectChanges();
+        expect(changeSpy).not.toHaveBeenCalled();
+        component.timeInput.hour = null;
+        component.timeInput.minute = null;
+        fixture.detectChanges();
+        dispatchFakeEvent(minuteEl, 'blur');
+        expect(changeSpy).toHaveBeenCalledTimes(1);
+      }));
+    });
```

Comment:
Actually the time picker does not currently have logic, since it is only taking the event emitted from the timeinput and emits it as output (which will be necessary for the datepicker later)  So it is now just using the timeinput and if range mode will be enabled for the future version, will have another timeinput.